feat: implement custom semantic release configuration with enhanced changelog and comments (#51)
pawcoding/astro-loader-pocketbase: .github/semantic-release.md

# Semantic Release Configuration

This repository uses a custom semantic release configuration to provide enhanced changelog generation and GitHub interaction.

## Features

### 📖 Enhanced Changelog

The changelog automatically categorizes commits with emojis and includes contributor information:

- ✨ **Features** - New functionality (`feat` commits)
- 🩹 **Fixes** - Bug fixes (`fix` commits)
- 📚 **Documentation** - Documentation updates (`docs` commits)
- 🛠️ **Internals** - Internal changes (`refactor`, `style`, `test`, `build`, `ci`, `chore` commits)
- ⚡ **Performance** - Performance improvements (`perf` commits)
- ↩️ **Reverts** - Reverted changes (`revert` commits)
- ⚠️ **BREAKING CHANGES** - Breaking changes (commits with `BREAKING CHANGE` footer)

Each commit entry includes:

- Scope (if provided)
- Commit message with linked issues and usernames
- Commit hash link
- Author attribution (@username)

### 💬 Custom GitHub Comments

#### Issue Comments

**For Pre-releases:**

```

> 🎉 This issue is included in version X.X.X-next.X which is now available for testing! 🚀

Reviewer: ```suggestion
🎉 This issue is included in version X.X.X-next.X which is now available for testing! 🧪
```

📦 NPM: astro-loader-pocketbase@X.X.X-next.X
📖 GitHub Release: X.X.X-next.X

> @username Would you mind testing this fix in your project and letting us know if it works as expected? We'd love your feedback!

Reviewer: ```suggestion
@username Can you check if everything works as expected in your project with this new version? Any feedback is welcome.
```

> This change will be included in the next regular release.

Reviewer: ```suggestion
<small>This change will be included in the next regular release.</small>
```

```

**For Regular Releases:**

```
🎉 This issue is included in version X.X.X which is now available! 🚀

📦 NPM: astro-loader-pocketbase@X.X.X
📖 GitHub Release: X.X.X
```

#### Pull Request Comments

**For Pre-releases:**

```

> 🎉 This PR is included in version X.X.X-next.X which is now available for testing! 🚀

Reviewer: ```suggestion
🎉 This PR is included in version X.X.X-next.X which is now available for testing! 🧪
```

📦 NPM: astro-loader-pocketbase@X.X.X-next.X
📖 GitHub Release: X.X.X-next.X

> @username Thank you for your contribution! This PR

Reviewer: ```suggestion
@username Thank you for your contribution!
```

> This change will be included in the next regular release.

Reviewer: ```suggestion
<small>This change will be included in the next regular release.</small>
```
